test/gpu/momentum_map.jl

```diff
@@ -0,0 +1,64 @@
+if isempty(GPUS)
+    @info """No functional GPUs found for testing, skipping tests...
+    To test GPU functionality, please use 'TEST_<GPU> = 1 julia ...' for one of GPU=[CUDA, AMDGPU, METAL, ONEAPI]"""
```

Comment:
In general, I like the implementation. However, maybe we could discuss changing the command-line interface. I think something like 

```bash
$ TEST_CUDA=1 julia -e "..."
$ TEST_CUDA=1 TEST_AMDGPU=1 julia -e "..."
```
Is not very convenient, especially if you test several platforms. Instead, I suggest to have the architecture as the value:

```bash
$ TEST_GPU=CUDA julia -e "..."
$ TEST_GPU=[CUDA, AMDGPU] julia -e "..."
```
This would also allow some convenient aliases, e.g. `AMDGPU==amdgpu` and alike.

Reply:
To be honest, I'm not sure if there are use cases. Normally you have only GPUs of one vendor in your system. Therefore I think this feature will be really seldom used.
In the CI, we will not use it, because each runner is has only GPUs of one vendor.

Reply:
And if nearly nobody is using the feature, I would avoid it because of the extra complexity of parsing, which is error prone.